internal/csi-addons/rbd/replication.go

```diff
@@ -308,17 +307,28 @@ func (rs *ReplicationServer) EnableVolumeReplication(ctx context.Context,
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 	if info.GetState() != librbd.MirrorImageEnabled.String() {
-		err = rbdVol.HandleParentImageExistence(ctx, flattenMode)
-		if err != nil {
-			log.ErrorLog(ctx, err.Error())
-
-			return nil, getGRPCError(err)
-		}
-		err = mirror.EnableMirroring(ctx, mirroringMode)
-		if err != nil {
-			log.ErrorLog(ctx, err.Error())
+		if len(volumes) > 0 {
+			for _, rbdVol := range volumes {
+				err = rbdVol.HandleParentImageExistence(ctx, flattenMode)
+				if err != nil {
+					err = fmt.Errorf("failed to handle parent image for volume group %q: %w", mirror, err)
+					return nil, getGRPCError(err)
+				}
+				err = mirror.EnableMirroring(ctx, mirroringMode)
```

Comment:
there are a lot of calls to this, why not combine it with the one of the `else` below?